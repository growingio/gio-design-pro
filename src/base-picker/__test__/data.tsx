import React from 'react';
import { Button } from '@gio-design/components';
import { PlusCircleFilled } from '@gio-design/icons';
import List from '../../list';

export const renderItems = () => (
  <>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
  </>
);

export const renderGroups = () => (
  <>
    <List.ItemGroup title="分组 1">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 2">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 3">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
  </>
);

export const footer = (
  <Button type="text" icon={<PlusCircleFilled />} size="small">
    新建内容
  </Button>
);

export const tabNavItems = [
  { key: 'tab-1', children: '选项 1' },
  { key: 'tab-2', children: '选项 2' },
];
