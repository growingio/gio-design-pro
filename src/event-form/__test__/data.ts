// eslint-disable-next-line import/prefer-default-export
export const spaceTags: any[] = [
  {
    id: 'JnG40OGz',
    name: '是',
    creator: 'haozhigang',
    creatorId: 'rRGoYQml',
    createdAt: '2021-01-14T07:08:25Z',
    updater: 'haozhigang',
    updaterId: 'rRGoYQml',
    updatedAt: '2021-01-14T07:08:25Z',
    description: '是  得到的',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/index/index',
      query: '得到的=s',
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'zZDb6aD9',
    name: '1月13，21次',
    creator: 'lvyuqiang',
    creatorId: 'n1QVaDyR',
    createdAt: '2021-01-13T09:03:46Z',
    updater: 'lvyuqiang',
    updaterId: 'n1QVaDyR',
    updatedAt: '2021-01-13T09:03:46Z',
    description: '1月13，21次',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/index/index',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'mgGJy2pA',
    name: 'wx123 的是页面 pages/form/form 。',
    creator: 'lvyuqiang',
    creatorId: 'n1QVaDyR',
    createdAt: '2021-01-13T07:14:25Z',
    updater: 'lvyuqiang',
    updaterId: 'n1QVaDyR',
    updatedAt: '2021-01-13T07:14:25Z',
    description: '现在定义的是页面 pages/form/form 。',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/form/form',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/form/form',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'AVpZWyGK',
    name: 'ls-xl',
    creator: 'liushuo',
    creatorId: 'qgQMnp3L',
    createdAt: '2021-01-12T05:14:29Z',
    updater: 'liushuo',
    updaterId: 'qgQMnp3L',
    updatedAt: '2021-01-12T05:14:29Z',
    description: '',
    platforms: ['web'],
    docType: 'elem',
    actions: ['chng'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/label/select.selection',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.title/form.basic-grey/label/select.selection',
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/07b18645-b784-4abe-b973-a08605bf5893.jpg?sign=wcyjkYl8LHTacSHpdojrXHeTrHPzHONADaFNg5OBY3U%3D&expires=1611037476177',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'WkDyn8Qe',
    name: 'ls-jk',
    creator: 'liushuo',
    creatorId: 'qgQMnp3L',
    createdAt: '2021-01-12T05:13:53Z',
    updater: 'liushuo',
    updaterId: 'qgQMnp3L',
    updatedAt: '2021-01-12T05:13:53Z',
    description: '',
    platforms: ['web'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '查找 query 柯基图片',
      index: null,
      href:
        'https://www.google.com/search?q=%E6%9F%AF%E5%9F%BA%E5%9B%BE%E7%89%87&oq=%E6%9F%AF%E5%9F%BA%E5%9B%BE%E7%89%87&aqs=chrome..69i57j0l2j69i61j69i60j0.4959j0j4&sourceid=chrome&ie=UTF-8',
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/a',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/a',
      index: null,
      href: null,
      content: '查找 query 柯基图片',
      pg: null,
      contentType: '=',
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/364c9f7b-9a27-47c5-94d2-bfe193e3c89a.jpg?sign=bFYchAI2ahfGZykwMOZL3Pm6NBYfc2srQPXu6jt8cC8%3D&expires=1611037476177',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'kqQe1qQr',
    name: '定义一个新的事件2',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:21:14Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:21:14Z',
    description: '',
    platforms: ['minp'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '主页',
      index: '0',
      href: 'pages/index/index',
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: '#onTabItemTap',
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: '12=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      pg: null,
      contentType: '=',
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: '7RDYqYDA',
    name: '事件2的页面',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:20:52Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:20:52Z',
    description: '',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: '12=12',
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'ExQWkgQ7',
    name: '事件4',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:19:40Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:19:40Z',
    description: '',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'xNQXjGMb',
    name: 'autotracker-cdp页面',
    creator: 'mayutong',
    creatorId: 'nxGK0Da6',
    createdAt: '2020-12-15T07:50:07Z',
    updater: 'mayutong',
    updaterId: 'nxGK0Da6',
    updatedAt: '2020-12-15T07:50:07Z',
    description: '',
    platforms: ['Android'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: '/TrackActivity',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'com.growingio.android.cdp',
      urlScheme: 'growing.447120b8e608d6b9',
      __typename: 'DocProps',
    },
    definition: {
      domain: 'com.growingio.android.cdp',
      path: '/TrackActivity',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: 'growing.447120b8e608d6b9',
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/2740844b-cb23-4868-8a43-d0f5cd150cc8.jpg?sign=LV5Clhvs36wRzmvx83nGR9J76ZvqDFp%2F1D54C7PixOs%3D&expires=1611037476178',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'nxGKXxDa',
    name: '验收限定条件2',
    creator: '马雨彤',
    creatorId: 'evp9bjGO',
    createdAt: '2021-02-08T03:34:14Z',
    updater: '马雨彤',
    updaterId: 'evp9bjGO',
    updatedAt: '2021-02-08T03:34:14Z',
    description: '不勾选',
    platforms: ['Android'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '加载了 并且初始化了 SDK',
      index: null,
      href: null,
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.title/div/div.title/h3',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: 'growing.314b5db864634231',
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.title/div/div.title/h3',
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: '=',
      urlScheme: 'growing.314b5db864634231',
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/ecfb4779-46d1-4a10-84b0-1818a9d721d8.jpg?sign=%2FCv2NaAELe7kyLCW7WRrJLckE%2F8X7QL%2FLx5A3FzdG6Y%3D&expires=1612852820607',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
];

export const deviceInfoMinp = { name: '小程序无埋点', domain: 'wx123456', os: 'minp' };
export const currentPageMinp = {
  id: 'f339fcb2bdd07fd7b55020c2835d8bede823e4af',
  eventType: 'PAGE',
  timestamp: 1610954652616,
  ip: '111.198.136.131',
  domain: 'wx123456',
  query: 'name=hello&a=3',
  platform: 'MinP',
  path: 'pages/index/index',
  appType: 0,
  actions: ['clck'],
};
export const deviceInfoApp = {
  projectId: '91eaf9b283361032',
  msgType: 'ready',
  timestamp: 1612755238635,
  domain: 'com.growingio.android.cdp',
  sdkVersion: '3.0.0',
  sdkVersionCode: 30000,
  os: 'Android',
  screenWidth: 1080,
  screenHeight: 1920,
  urlScheme: 'growing.314b5db864634231',
  name: 'Android无埋点测试',
};
export const currentPageApp = {
  path: '/MainActivity',
  title: 'AutoTracker-CDP',
  left: 0,
  top: 0,
  width: 343.125,
  height: 610,
  appType: 1,
  domain: 'com.growingio.android.cdp',
  isDefined: false,
};
