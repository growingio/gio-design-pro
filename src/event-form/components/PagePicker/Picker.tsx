/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable @typescript-eslint/no-use-before-define */
import React, { useEffect, useRef, useState } from 'react';
import { Button, SearchBar } from '@gio-design/components';
import './index.less';
import { DownFilled, PlusCircleFilled } from '@gio-design/icons';
import useUpdate from '../../../hooks/useUpdate';
import { isChildOfElement } from '../../utils';
import PageList from './PageList';
// import PlusCircleFilled from '@gio-design/icons/es/PlusCircleFilled';
import { Element as TagElement } from '../../../types';

interface Prop {
  // 去定义定的页面
  onDefineNewPage: (e?: any) => void;
  currentPageTags: TagElement[];
  relatedPageTags: TagElement[];
  onSelect?: (selectedPageTag: TagElement) => void;
}

// 失去焦点是 判断当前页面为空 提示错误
const PagePicker: React.FC<Prop> = (props) => {
  const { currentPageTags, relatedPageTags, onSelect, onDefineNewPage } = props;

  const root = useRef<any>();
  const [expand, setExpand] = useState(false);
  const [searchText, updateSearchText] = useState('');
  const [value, setValue] = useState<TagElement>();
  const [hasError] = useState(false);

  useEffect(() => {
    // eslint-disable-next-line @typescript-eslint/no-use-before-define
    document.body.addEventListener('click', cancelExpand);
    return () => {
      document.body.removeEventListener('click', cancelExpand);
    };
  }, []);

  useEffect(() => {
    const page = currentPageTags?.[0];
    if (page) {
      onSelect?.(page);
    }
    setValue(page);
  }, [currentPageTags]);

  useUpdate(() => {
    let error = false;
    if (!expand && !value) {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      error = true;
    }
    // 不提示页面选择的error
    // updateError(error);
  }, [expand, value]);

  /**
   * 隐藏页面选中面板
   */
  const cancelExpand = (e: any) => {
    if (!isChildOfElement(e.target, root.current)) {
      setExpand(false);
      updateSearchText('');
    }
  };

  /**
   * 搜索文本变化
   * @param nv 文本值
   */
  const onSearchBarChange = (nv: any) => {
    updateSearchText(nv);
  };

  /**
   * 页面列表选择器选中响应
   * @param tag  在相似页面中选中的tag
   * @param index
   */
  const onPageListSelected = (tag: TagElement) => {
    onSelect?.(tag);
    setValue(tag);
    setExpand(false);
  };

  /**
   * 把tag装换为显示的值
   * @param tag
   */
  const tagToValue = (tag: TagElement) => {
    const isCurrent = currentPageTags?.findIndex((v) => v.id === tag.id) !== -1;
    const { path } = tag.definition;
    let text = `${tag.name} | ${tag.definition.domain}${path || '下的所有页面'}`;
    // 是小程序
    if (tag.platforms.includes('minp')) {
      text = `${tag.name} | ${path || '小程序所有页面'}`;
    }
    return (
      <>
        {isCurrent && (
          <span
            style={{
              color: '#1248e9',
              marginRight: 4,
            }}
          >
            [当前页]
          </span>
        )}
        <span>{text}</span>
      </>
    );
  };

  return (
    <div ref={root} className={`page-selector ${hasError ? 'page-selector-error' : ''}`}>
      <div className={`page-selector-input ${expand ? 'focus' : ''}`} onClick={() => setExpand(!expand)}>
        {value ? tagToValue(value) : <span className="placeholder">请选择所属页面</span>}
        <DownFilled className="down-icon" color="#abb2c3" style={{ transform: expand ? 'rotate(180deg)' : 'none' }} />
      </div>
      {hasError && <span className="page-selector-error-text">所属页面不能为空</span>}
      {expand && (
        <div className="page-selector-dropdown">
          <div className="page-search-box">
            <SearchBar
              id="page-search"
              placeholder="搜索页面名称"
              value={searchText}
              onChange={onSearchBarChange}
              showClear
            />
          </div>
          <PageList
            value={value}
            tags={relatedPageTags.filter((v) => v.name.includes(searchText))}
            labelRenderer={tagToValue}
            inSearch={!!searchText}
            onSelect={onPageListSelected}
          />
          <div className="page-selector-footer">
            <Button size="small" type="text" icon={<PlusCircleFilled />} onClick={onDefineNewPage}>
              <span style={{ marginLeft: 8 }}>定义新页面</span>
            </Button>
          </div>
        </div>
      )}
    </div>
  );
};

export default PagePicker;
