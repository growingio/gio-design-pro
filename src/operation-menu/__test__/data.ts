export const options = [
  {
    disabled: false,
    tooltip: '',
    value: 'update-space-role',
    label: '修改站点角色',
  },
  {
    disabled: false,
    tooltip: '',
    value: 'update-space',
    label: '删除用户',
  },
  {
    disabled: true,
    tooltip: '行行行',
    value: 'update-spxxxace',
    label: '删除用户2',
  },
];

export default {
  options,
};
