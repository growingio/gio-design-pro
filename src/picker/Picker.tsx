import * as React from 'react';
import Alert from '@gio-design-new/components/es/components/alert';
import SearchBar from '@gio-design-new/components/es/components/search-bar';
import Loading from '@gio-design-new/components/es/components/loading';
import TabNav from '@gio-design-new/components/es/components/tab-nav';
import Button from '@gio-design-new/components/es/components/button';
import Menu from '@gio-design-new/components/es/components/cascader/menu';
import Input from '@gio-design-new/components/es/components/input';
import Dropdown from '@gio-design-new/components/es/components/dropdown';
import usePrefixCls from '@gio-design-new/components/es/utils/hooks/use-prefix-cls';
import { UpFilled, DownFilled } from '@gio-design/icons';
import { NodeData } from '@gio-design-new/components/es/components/cascader/interface';
import EmptyPrompt from '../empty-prompt';
import { PickerProps } from './interfaces';

function Picker({
  visible = false,
  onVisibleChange,
  inputValue,
  input,
  searchPlaceholder = 'Search label',
  onSearch,
  tabNav,
  loading = false,
  groupVisible = true,
  dataSource,
  actionButton,
  onHoverPanelShow,
  onSelect,
  placeholder = '请选择',
  value,
  emptyPrompt = { description: 'No Data' },
}: PickerProps) {
  const prefixCls = usePrefixCls('picker');
  const [query, setQuery] = React.useState<string>('');

  function onQueryChange(current: string) {
    setQuery(current);
    onSearch?.(current.slice(0, 200).toLocaleLowerCase());
  }

  function onMenuHover(current: NodeData) {
    return !onHoverPanelShow ? null : <div className={`${prefixCls}-hover-panel`}>{onHoverPanelShow(current)}</div>;
  }

  function renderList() {
    if (loading) {
      return (
        <div className="gio-loading-wrapper">
          <Loading />
        </div>
      );
    }

    if (dataSource.length === 0) {
      return <EmptyPrompt {...emptyPrompt} />;
    }

    return (
      <Menu
        groupName={groupVisible}
        className={`${prefixCls}__list`}
        dataSource={dataSource}
        onSelect={onSelect}
        afterInner={onMenuHover}
        value={value}
      />
    );
  }
  function renderOverlay() {
    return (
      <div className={prefixCls}>
        <div className={`${prefixCls}__header`}>
          <SearchBar placeholder={searchPlaceholder} size="middle" value={query} onChange={onQueryChange} />
          {actionButton && <Button type="secondary" {...actionButton} />}
        </div>
        <>
          {query.length > 200 && (
            <Alert type="warning" message="搜索字符长度已超过200，只取前200字符搜索" size="small" showIcon closeable />
          )}
        </>
        {tabNav && (
          <TabNav type="line" size="small" onChange={tabNav.onChange}>
            {React.useMemo(() => tabNav.items.map((i) => <TabNav.Item {...i} />), [tabNav.items])}
          </TabNav>
        )}
        {renderList()}
      </div>
    );
  }
  const defaultTrigger = (
    <Input readOnly placeholder={placeholder} value={inputValue} suffix={visible ? <UpFilled /> : <DownFilled />} />
  );
  return (
    <Dropdown
      visible={visible}
      onVisibleChange={onVisibleChange}
      placement="bottomLeft"
      overlayClassName={`${prefixCls}-dropdown`}
      overlay={renderOverlay}
    >
      <div className={`${prefixCls}-trigger`}>
        {/* <Input
          placeholder={placeholder}
          aria-label="picker-value"
          suffix={visible ? <UpFilled /> : <DownFilled />}
          value={inputValue}
        /> */}
        {React.isValidElement(input) ? input : defaultTrigger}
      </div>
    </Dropdown>
  );
}

export default Picker;
