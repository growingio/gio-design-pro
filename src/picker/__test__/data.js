export const nodes = [
  {
    label: 'Data 1',
    value: 'data-1',
  },
  {
    label: 'Data 2',
    value: 'data-2',
  },
  {
    label: 'Data 3',
    value: 'data-3',
  },
];

export default {
  nodes,
};
