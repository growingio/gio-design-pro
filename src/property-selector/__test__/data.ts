// eslint-disable-next-line import/prefer-default-export
export const insightDimensions = [
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'd',
    name: '域名',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_t_date_1',
    name: '测试日期型',
    type: 'usr',
    valueType: 'date',
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_t_number_1',
    name: '测试数值型',
    type: 'usr',
    valueType: 'int',
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_sex002',
    name: 'sex002',
    type: 'usr',
    valueType: 'string',
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'rd',
    name: '访问来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'rt',
    name: '一级访问来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'kw',
    name: '搜索词',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_source',
    name: '广告来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_campaign',
    name: '广告名称',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_content',
    name: '广告内容',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_term',
    name: '广告关键字',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_medium',
    name: '广告媒介',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'ch',
    name: '自定义App渠道',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'city',
    name: '城市',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'region',
    name: '地区',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'countryCode',
    name: '国家代码',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'countryName',
    name: '国家名称',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'bw',
    name: '浏览器',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'bwv',
    name: '浏览器版本',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'os',
    name: '操作系统',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'osv',
    name: '操作系统版本',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'shw',
    name: '屏幕大小（高*宽）',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'l',
    name: '操作系统语言',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'db',
    name: '设备品牌',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'dm',
    name: '设备型号',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'o',
    name: '设备方向',
    type: 'global',
    valueType: null,
  },
  {
    id: 'AbQ3bDYe',
    isPrimaryKey: false,
    key: 'order_name',
    type: 'itm',
    groupId: 'item',
    name: 'shangpin',
    groupName: '物品属性',
    valueType: 'string',
    description: '',
    associatedId: 'var_engageType',
  },
  {
    associatedId: '',
    groupId: 'event',
    groupName: '事件变量',
    id: 'var_engageType',
    name: '产品交互性参与类型',
    type: 'event',
    valueType: 'string',
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'tm',
    name: '时间我是超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'b',
    name: '应用平台',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'cv',
    name: '应用版本',
    type: 'global',
    valueType: null,
  },
];
