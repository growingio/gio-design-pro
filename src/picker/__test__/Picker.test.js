import React from 'react';
import ReactDOM from 'react-dom';
// import { PlusOutlined } from '@gio-design/icons';
import { mount, render } from 'enzyme';
import Picker from '..';
import { nodes } from './data';

const onVisibleChange = jest.fn();
const defaultProps = {
  dataSource: [],
  onVisibleChange,
  inputValue: '',
};

describe('Picker', () => {
  it('can search in dropdown', () => {
    const onSearch = (query) => {
      expect(query).not.toBeNull();
      expect(query.length).toBeLessThanOrEqual(200);
    };
    const wrapper = mount(
      <Picker {...defaultProps} visible dataSource={nodes} onHoverPanelShow={jest.fn()} onSearch={onSearch} />
    );
    const search = wrapper.find('.gio-picker-dropdown .gio-input-content');
    search.simulate('change', { target: { value: 'x' } });

    search.simulate('change', { target: { value: 'x'.repeat(201) } });
    expect(wrapper.find('.gio-alert')).toHaveLength(1);
  });
});

describe('Picker snapshots', () => {
  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  it('will only show input', () => {
    expect(render(<Picker {...defaultProps} />)).toMatchSnapshot();
  });

  // it('will show input with empty dropdown', () => {
  //   expect(render(<Picker {...defaultProps} visible />)).toMatchSnapshot();
  // });

  // it('will show input with loading dropdown', () => {
  //   expect(render(<Picker {...defaultProps} visible loading />)).toMatchSnapshot();
  // });

  // it('will show input with dropdown', () => {
  //   const component = (
  //     <Picker
  //       {...defaultProps}
  //       visible
  //       dataSource={nodes}
  //       onHoverPanelShow={jest.fn()}
  //       tabNav={{ items: [{ key: 'all', label: 'All' }] }}
  //       actionButton={{ icon: <PlusOutlined />, onClick: jest.fn() }}
  //     />
  //   );
  //   expect(render(component)).toMatchSnapshot();
  // });
});
