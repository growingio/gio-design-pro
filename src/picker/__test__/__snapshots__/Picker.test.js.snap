// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker snapshots will only show input 1`] = `
<div
  class="gio-picker-trigger"
  style="cursor:pointer"
>
  <div
    class="gio-input gio-input--middle gio-input__container"
  >
    <input
      aria-label="picker-value"
      class="gio-input__content gio-input__content-suffix"
      placeholder=""
      type="text"
      value=""
    />
    <div
      class="gio-input__suffix"
    >
      <span
        class="gio-icon"
      >
        <svg
          class="gio-icon-svg"
          fill="currentColor"
          height="16px"
          viewBox="0 0 14 14"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <g
            data-name="\\\\u56FE\\\\u5C42 2"
            id="down-filled_svg__\\\\u56FE\\\\u5C42_2"
          >
            <g
              data-name="\\\\u56FE\\\\u5C42 1"
              id="down-filled_svg__\\\\u56FE\\\\u5C42_1-2"
            >
              <path
                d="M7.39 9.51a.5.5 0 01-.78 0l-3-3.7a.51.51 0 01-.06-.53A.5.5 0 014 5h6a.5.5 0 01.45.28.51.51 0 01-.06.53z"
                fill="currentColor"
                fill-rule="evenodd"
                id="down-filled_svg__down-2"
              />
            </g>
          </g>
        </svg>
      </span>
    </div>
  </div>
</div>
`;
